carnival/carnival.ts

```diff
@@ -0,0 +1,1017 @@
+            protected newBest?: boolean
+        ) {
+            super(screen.width, 46, img`
+        1 1 1
+        f f f
+        1 1 1
+        `);
+            this.cursorOn = false;
+
+            /* Since best time is lower, need to do this elsewhere
+            this.isNewbestScore = this.score > this.bestScore;
+            */
+        }
+
+        displayCursor() {
+            this.cursorOn = true;
+        }
+
+        update() {
+            this.clearInterior();
+            this.drawTextCore();
+
+            if (this.cursorOn) {
+                this.drawCursorRow();
+            }
+        }
+
+        drawTextCore() {
+            const titleHeight = 8;
+
+            this.image.printCenter(
+                this.messageText,
+                titleHeight,
+                screen.isMono ? 1 : 5,
+                image.font8
+            );
+
+
+            if (this.score != undefined) {
+                const scoreHeight = 23;
+                const bestScoreHeight = 34;
+                const scoreColor = screen.isMono ? 1 : 2;
+
+                this.image.printCenter(
+                    "Score:" + this.score,
+                    scoreHeight,
+                    scoreColor,
+                    image.font8
+                );
+
+                if (this.newBest == true) {
+                    this.image.printCenter(
+                        "New Best Score!",
+                        bestScoreHeight,
+                        scoreColor,
+                        image.font5
+                    );
+                } else {
+                    this.image.printCenter(
+                        "Best:" + this.best,
+                        bestScoreHeight,
+                        scoreColor,
+                        image.font8
+                    );
+                }
+            }
+        }
+    }
+
+    // End Info
+
+    // Formerly namespace Game
+
+
+    /**
+     * Adds additional end game styles
+     */
+    //% color="#b70082"
+    //% group="Game"
+    //% blockId=on_game_over_expanded
+    //% block="game over $winStyle || add effect $winEffect"
+    //% winType.defl=winTypes.Win
+    //% winEffect.defl=effects.confetti
+    //% inlineInputMode=inline
+    //% help=github:carnival/docs/on_game_over_expanded
+    export function onGameOverExpanded(winStyle: winTypes, winEffect?: effects.BackgroundEffect) {
+
+        if (winEffect == undefined) {
+            if (winStyle == winTypes.Lose) { winEffect = effects.melt; }
+            else { winEffect = effects.confetti; }
+        }
+
+        if (winStyle == winTypes.Lose) {
+            game.over(false, winEffect)
+        } else {
+            newGameOver(winStyle, winEffect);
+        }
+    }
+
+    /**
+     * Adds custom end game styles
+     */
+    //% color="#b70082"
+    //% group="Game"
+    //% blockId=on_game_over_custom_expanded
+    //% block="game over $message || with $winEffect and $gameSound send score $score judge $scoring"
```

Comment:
the wording of this block seems a little confusing. what is "send score" referring to? and "judge"?

Reply:
I wish I could figure out how to do this without making it so long.  How about:

    //% block="game over $message || with $winEffect and $gameSound judged as $scoring using score $score"

The idea is that they can choose whether the winner has highest/lowest/multiplayer…and optionally pass in their own "score" if it's not a time or points system.